benchexec/tooladapter.py

```diff
@@ -0,0 +1,88 @@
+# This file is part of BenchExec, a framework for reliable benchmarking:
+# https://github.com/sosy-lab/benchexec
+#
+# SPDX-FileCopyrightText: 2007-2020 Dirk Beyer <https://www.sosy-lab.org>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+"""Utilities for adapting older tool-info modules to the currently expected API."""
+
+import inspect
+
+from benchexec.tools.template import BaseTool, BaseTool2
+
+import benchexec.model
+
+
+CURRENT_BASETOOL = BaseTool2
+"""Alias for the latest base-tool class in benchexec.tools.template"""
```

Comment:
Developers of tool-info modules may be tempted to use this constant as type.
I would explicitly discourage the use of this constant in tool-info modules, and tell users in the pydoc that they should use the concrete `BaseToolX` type

Reply:
Hm, isn't it enough that this is in a completely different module that tool-info modules should not even import, and developers of tool-info modules should not even know about typically?

We could add such a general comment to `benchexec.tooladapter`.

Reply:
Probably. A clarifying comment can still be added later, if it actually is used wrong.